addons/mail/tests/common.py

```diff
@@ -1403,33 +1407,31 @@ def assertBusNotifications(self, channels, message_items=None, check_unique=True
         """
         self.env.cr.precommit.run()  # trigger the creation of bus.bus records
         bus_notifs = self.env['bus.bus'].sudo().search([('channel', 'in', [json_dump(channel) for channel in channels])])
-        new_line = "\n"
+        new_lines = "\n\n"
 
         def notif_to_string(notif):
-            message = json.loads(notif.message)
-            payload = json_dump(message.get("payload"))
-            return f"{notif.channel}  # {message.get('type')} - {payload[0:120]}{'…' if len(payload) > 120 else ''}"
+            return f"{notif.channel}\n{notif.message}"
 
         self.assertEqual(
             bus_notifs.mapped("channel"),
             [json_dump(channel) for channel in channels],
-            f"\nExpected:\n{new_line.join([json_dump(channel) for channel in channels])}"
-            f"\nReturned:\n{new_line.join([notif_to_string(notif) for notif in bus_notifs])}",
+            f"\n\nExpected:\n{new_lines[0].join([json_dump(channel) for channel in channels])}"
+            f"\n\nReturned:\n{new_lines.join([notif_to_string(notif) for notif in bus_notifs])}",
         )
-        notif_messages = [n.message for n in bus_notifs]
         for expected in message_items or []:
-            for notification in notif_messages:
-                if json_dump(expected) == notification:
+            for notification in bus_notifs:
+                if json.loads(json_dump(expected)) == json.loads(notification.message):
```

Comment:
Note: change/fix here. Exact json string match was overkill, only semantics/actual data should be checked.